modules/tensor_mechanics/src/materials/CZM/CZM_3DCMaterial.C

```diff
@@ -0,0 +1,166 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "Assembly.h"
+#include "CZM_3DCMaterial.h"
+#include "RotationMatrix.h"
+
+registerMooseObject("TensorMechanicsApp", CZM_3DCMaterial);
+
+template <>
+InputParameters
+validParams<CZM_3DCMaterial>()
+{
+  InputParameters params = validParams<CZMMaterialBase>();
+  params.addClassDescription("3DC cohseive law model, no damage");
+  params.addRequiredParam<std::vector<Real>>(
+      "DeltaU0",
+      "a vector containing the displacement value at which maximum"
+      " traction occurs for the normal(1st) and tangential(2nd) "
+      " direction.");
+  params.addRequiredParam<std::vector<Real>>("MaxAllowableTraction",
+                                             "a vector containing the maximum allowed traction"
+                                             "for the normal(1st) and tangential(2nd) direction.");
+  params.addClassDescription("Simple Exponential cohseive law model, with damage");
+  return params;
+}
+
+CZM_3DCMaterial::CZM_3DCMaterial(const InputParameters & parameters)
+  : CZMMaterialBase(parameters),
+    _deltaU0(getParam<std::vector<Real>>("DeltaU0")),
+    _maxAllowableTraction(getParam<std::vector<Real>>("MaxAllowableTraction"))
+
```

Comment:
is this bilinear law? if so, you can rename the class to CZMBilinearLaw?

Reply:
No this is from [Salehani et al.](https://arxiv.org/pdf/1801.03430.pdf)